File: Java/ShellSort.java
package Java;

public class ShellSort extends Sort{
	
	public static void main(String[] args){
		int[] data = buildDataSet(5);
		
		timeSort(new ShellSort(), data);
	}
	
	public ShellSort(){ super("Shell"); }
	
	private void shellSort(int[] data){
	     for( int gap = data.length/2; gap > 0; gap /= 2 ) // determine sequence
	         for( int i = gap; i < data.length; i++ )
	         {
	             int temp = data[i]; // create hole
	             
	             //slide greater elements down
	    	     int j;
	             for(j = i; j >= gap && data[j-gap] > temp; j -= gap )
	            	 data[j] = data[j-gap];
	             
	             data[j] = temp;
	         }
	}
	
	@Override
	public void sort(int[] data){	     
		shellSort(data);
	}
}

File: javascript/shell-sort.js
function shellSort(array){
    for(let gap = Math.floor(array.length/2); gap > 0; gap /= 2) // determine sequence
        for(let i = Math.floor(gap); i < array.length; i++)
        {
            let temp = array[i]; // create hole
            
            //slide greater elements down
            var j;
            for(j = i; j >= gap; j -= gap )
            {	 	            	 
                if ( array[j-gap] > temp )
                    array[j] = array[j-gap];
                else
                    break;
            }
            array[j] = temp;
        }
}

function sort(array){    
    shellSort(array);
}